File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if(len(needle) > len(haystack)):
            return -1
        len_needle = len(needle)
        for i in range(len(haystack)):
            if(haystack[i:i+len_needle] == needle):
                return i
        return -1


        

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if(needle.length != haystack.length){
        let n = needle.length - 1
    let left = 0
    while(left <= haystack.length - n - 1){
        let window = haystack.slice(left, left+n+1)
        if(window == needle) return left
        left++
    }
    return -1
    }
    else{
        if(haystack == needle) return 0
        return -1
    }
    
};